armi/reactor/converters/uniformMesh.py

```diff
@@ -244,6 +256,11 @@ def _buildAllUniformAssemblies(self):
             newAssem = self.makeAssemWithUniformMesh(sourceAssem, self._uniformMesh)
             src = sourceAssem.spatialLocator
             newLoc = self.convReactor.core.spatialGrid[src.i, src.j, 0]
+            runLog.debug(
```

Comment:
Is this debug statement needed/useful for any particular reason?

Reply:
No, I was using it but we probably shouldn't keep it around. There were some cases where it was taking ~20 minutes to convert a third core reactor to full core, and this was the only log message that would let me know the code was still running and not frozen during that period. But it does generate a lot of output and is probably not necessary.

Reply:
That's a long time. Is that still happening? Maybe the loop is inefficient if so

Reply:
In a recent run, some were taking 4 minutes and others about 17 min:
```
> grep "Reactor core conversion" *.stdout
[xtra] Reactor core conversion time: 248.86255460200005 seconds
[xtra] Reactor core conversion time: 244.63209826000002 seconds
[xtra] Reactor core conversion time: 247.14147353800013 seconds
[xtra] Reactor core conversion time: 246.28612149000037 seconds
[xtra] Reactor core conversion time: 818.0751615429999 seconds
[xtra] Reactor core conversion time: 1006.6690055039999 seconds
[xtra] Reactor core conversion time: 1036.225465257 seconds
[xtra] Reactor core conversion time: 1056.5681029470006 seconds
[xtra] Reactor core conversion time: 1057.0175631559996 seconds
[xtra] Reactor core conversion time: 1062.2484797609995 seconds
[xtra] Reactor core conversion time: 1067.229568883 seconds
[xtra] Reactor core conversion time: 1043.9269293489997 seconds
[xtra] Reactor core conversion time: 1061.6862967769994 seconds
[xtra] Reactor core conversion time: 1000.8848847240006 seconds
[xtra] Reactor core conversion time: 1016.5796172129994 seconds
[xtra] Reactor core conversion time: 1052.6184727269992 seconds
```

Reply:
So, this change to the code makes it takes 20 minutes to convert a 1/3 core reactor to a full core?

But it normally only takes seconds, right?

If I understand the situation correctly, we can't let this PR be merged until we have fixed this issue.  People would be very upset with us.

Reply:
The reactor core conversion was taking 20 minutes before I started this PR. I have no idea when or why that started happening, but it's not from this code change.

Reply:
Hm. When I profiled the core it the major slow downs were due to setting the number densities on the blocks because we reach down the composite chain to derive volumes each time and these aren't being cached. We used to have caching but that was removed because it caused a ton of issues. I think that if we could speed up the set number densities call then this would speed up significantly @john-science.

Reply:
IMO, since this slowdown predates this PR, I vote we make a new issue to address the number density calculation and get this PR merged. Just my two cents.

Reply:
I agree - Just reviewed and approved. Ill make a general ticket on number density speed ups

Reply:
https://github.com/terrapower/armi/issues/721

Reply:
Thanks all. I didn't realize quite the extent of how long this was taking (before this PR).

I can't help but think that is a solvable problem.

Reply:
> Thanks all. I didn't realize quite the extent of how long this was taking (before this PR).
> 
> I can't help but think that is a solvable problem.

100%.